Makefile

```diff
@@ -1,71 +1,76 @@
 .DEFAULT_GOAL := all
 sources = pydantic tests docs/plugins
 
+.PHONY: .pdm
+.pdm:  ## Check that PDM is isntallced
+	@pdm -V || echo 'Please install PDM: https://pdm.fming.dev/latest/\#installation'
+
+.PHONY: .pre-commit
+.pre-commit:  ## Check that pre-commit is installed
+	@pre-commit -V || echo 'Please install pre-commit: https://pre-commit.com/'
+
 .PHONY: install
-install:
-	python -m pip install -U pip
-	pip install -r requirements/all.txt
-	pip install -r requirements/pyproject-all.txt
-	pip install -e .
+install: .pdm .pre-commit
+	pdm install --group :all
+	pre-commit install --install-hooks
 
 .PHONY: refresh-lockfiles
-refresh-lockfiles:       ## Sync lockfiles with requirements files.
-	bash requirements/refresh.sh
-
+refresh-lockfiles: .pdm       ## Sync lockfiles with requirements files.
+	pdm lock --refresh --dev --group :all
 
 .PHONY: rebuild-lockfiles
-rebuild-lockfiles:       ## Rebuild lockfiles from scratch, updating all dependencies
-	bash requirements/rebuild.sh
+rebuild-lockfiles: .pdm       ## Rebuild lockfiles from scratch, updating all dependencies
+	pdm lock --dev --group :all
 
 .PHONY: format
-format:
-	black $(sources)
-	ruff --fix $(sources)
+format: .pdm
```

Comment:
I don't think we need ` .pdm` on this or later commands - we don't check everything else is installed.

If you do want this behaviour, maybe remove `PHONY` from the `.pdf` recipe and make will check it's been run without calling `pdm`.